patches/tModLoader/Terraria/Utils.cs.patch

```diff
@@ -34,6 +50,49 @@
  						if (num3 < 0)
  							num3 = 0;
  
+@@ -645,21 +_,30 @@
+ 		}
+ 	}
+ 
++	private static void OpenFolderXdg(string folderPath) {
++		ProcessStartInfo processStartInfo = new ProcessStartInfo {
++			FileName ="xdg-open",
++			Arguments = folderPath,
++			UseShellExecute = true,
++			CreateNoWindow = true
++		};
++
++		processStartInfo.EnvironmentVariables["LD_LIBRARY_PATH"] = "/usr/lib:/lib";
++		Process.Start(processStartInfo);
++	}
++
+ 	public static void OpenFolder(string folderPath)
+ 	{
+-		if (TryCreatingDirectory(folderPath)) {
+-			if (Platform.IsLinux) {
+-				Process.Start(new ProcessStartInfo(folderPath) {
+-					FileName = "open-folder",
+-					Arguments = folderPath,
+-					UseShellExecute = true,
+-					CreateNoWindow = true
+-				});
+-			}
+-			else {
+-				Process.Start(new ProcessStartInfo(folderPath) { UseShellExecute = true });
+-			}
++		if (!TryCreatingDirectory(folderPath))
++			return;
++
++		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ||
++			RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) {
++			Process.Start(new ProcessStartInfo(folderPath) { UseShellExecute = true });
++			return;
+ 		}
++
++		OpenFolderXdg(folderPath);
```

Comment:
Please try and minimise patch size. Comment out the vanilla `Process.Start` lines under the linux conditional, and insert `OpenFolderXdg(folderPath)` directly after the comment. 

https://github.com/tModLoader/tModLoader/wiki/tModLoader-Style-Guide#comment-out-vanilla-code-rather-than-removing-it

I've noticed in your other recent commit that you've been considering the `FreeBSD` platform. While I appreciate the forethinking, tML doesn't currently run on FreeBSD and we haven't had any requests for it, so it's better for maintainability to leave the structure of code as-is. 

`ReLogic.Platform.IsLinux` also covers FreeBSD at the moment

Reply:
Sorry for the formatting errors. I was not aware of these guidelines.